cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/LoadPersistedConcepts.kt

```diff
@@ -0,0 +1,318 @@
+        underlyingNode.conceptBuildHelper(
+            name = concept.name,
+            underlyingNode = underlyingNode,
+            constructorArguments =
+                concept.constructorArguments?.associate { arg -> arg.name to arg.value }
+                    ?: emptyMap(),
+            connectDFGUnderlyingNodeToConcept =
+                concept.dfg?.fromThisNodeToConcept ?: false, // TODO: this `?: false` is not nice
+            connectDFGConceptToUnderlyingNode =
+                concept.dfg?.fromConceptToThisNode ?: false, // TODO: this `?: false` is not nice
+        )
+    }
+
+    /** The root node of our YAML/JSON structure. It contains a list of [ConceptRootEntry]s. */
+    private data class PersistedConceptsEntry(val concepts: List<ConceptRootEntry>?)
+
+    /**
+     * This class represents a single entry in the YAML/JSON file. It contains the concept and
+     * optionally a location and/or a signature entry. The pass itself enforces that exactly one of
+     * location or signature is set.
+     *
+     * @param concept The [ConceptEntry] containing the concept information.
+     * @param location The [LocationEntry] containing the [PhysicalLocation] information to match
+     *   against.
+     * @param signature The [SignatureEntry] containing the signature of nodes to match.
+     */
+    private data class ConceptRootEntry(
+        val concept: ConceptEntry,
+        val location: LocationEntry?,
+        val signature: SignatureEntry?,
+    )
+
+    /**
+     * This class represents a single concept entry in the YAML/JSON file. It contains the name of
+     * the concept, optional constructor arguments, and optional DFG connections.
+     *
+     * @param name The FQN of the concept to be added.
+     * @param constructorArguments The constructor arguments to be passed to the concepts'
+     *   constructor.
+     * @param dfg The DFG connections to be created between the concept and the underlying node.
+     */
+    private data class ConceptEntry(
+        val name: String,
+        val constructorArguments: List<ConstructorArgumentEntry>?,
+        val dfg: DFGEntry?,
+    )
+
+    /**
+     * This class represents the DFG connections to be created between the concept and the
+     * underlying node.
+     *
+     * @param fromThisNodeToConcept Whether to add a DFG connection from the underlying node to the
+     *   concept node.
+     * @param fromConceptToThisNode Whether to add a DFG connection from the concept node to the
+     *   underlying node.
+     */
+    private data class DFGEntry(
+        val fromThisNodeToConcept: Boolean?,
+        val fromConceptToThisNode: Boolean?,
+    )
+
+    /**
+     * This class represents a single constructor argument entry in the YAML/JSON file.
+     *
+     * @param name The name of the constructor argument.
+     * @param value The value of the constructor argument.
+     */
+    private data class ConstructorArgumentEntry(val name: String, val value: String)
```

Comment:
Can this be aligned with the code used in #2183? It looks like at least the ConstructorArguments are somehow redundant.

Reply:
Well, in both cases we store similar/the same(?) information, so I thought we could maybe merge the two classes. Also, when we generate a yaml file later, this may be useful

Reply:
Slightly different syntax and we need to store more information in the other one such as the type, etc. Let's keep them separate and maybe harmonize later.